### DISCUSSION TITLE
Cask 'mysqlworkbench' definition is invalid: Token '{:v1=>"mysqlworkbench"}' in header line does not match the file name.

### DISCUSSION THREAD

**Discussion by saulspatz** [Homebrew/discussions#407] [Casks] [answered]
When I run `brew upgrade` I get the message

    Error: Cask 'mysqlworkbench' definition is invalid: Token '{:v1=>"mysqlworkbench"}' in header line does not match the file name.

I haven't had mysqlworkbench on my computer for a long time, and this is the first time I've ever seen this message.  I can't think what can have changed.  The only thing of significance I can think of is that homebrew upgraded me from python 3.8 to python 3.9 today, and that kicked off some dependency checks.   

When I `ls -l /usr/local/Cellar`, mysqlworkbench is not shown.  I have tried `brew cleanup`, `brew uninstall --force mysqlworkbench`, and `brew uninstall --force 'mysqlworkbench'` but I always get the same error when I run `brew upgrade.`  I have considered reinstalling mysqlworkbench and then uninstalling it, but I'm afraid it might just make things worse, since I don't understand what's going on.

Here's the output from `brew config`:

    HOMEBREW_VERSION: 2.7.1-68-ga7b61c6
    ORIGIN: https://github.com/Homebrew/brew
    HEAD: a7b61c645a2a95ef888847e8b4d4b017fa9b745f
    Last commit: 4 hours ago
    Core tap ORIGIN: https://github.com/Homebrew/homebrew-core
    Core tap HEAD: 79f360462ac6afac848bd7caa2e13169145da7b6
    Core tap last commit: 3 hours ago
    Core tap branch: master
    HOMEBREW_PREFIX: /usr/local
    HOMEBREW_CASK_OPTS: []
    HOMEBREW_DISPLAY: /private/tmp/com.apple.launchd.pIuzQpeKXj/org.macosforge.xquartz:0
    HOMEBREW_MAKE_JOBS: 4
    Homebrew Ruby: 2.6.3 => /usr/local/Homebrew/Library/Homebrew/vendor/portable-ruby/2.6.3_2/bin/ruby
    CPU: quad-core 64-bit ivybridge
    Clang: 12.0 build 1200
    Git: 2.30.0 => /usr/local/bin/git
    Curl: 7.64.1 => /usr/bin/curl
    macOS: 10.15.7-x86_64
    CLT: 12.3.0.0.1.1607026830
    Xcode: 12.3
    XQuartz: 2.7.11 => /opt/X11

`brew doctor` gives me:

    Please note that these warnings are just used to help the Homebrew maintainers
    with debugging if you file an issue. If everything you use Homebrew for is
    working fine: please don't worry or file an issue; just ignore this. Thanks!
    
    Warning: Unbrewed header files were found in /usr/local/include.
    If you didn't put them there on purpose they could cause problems when
    building Homebrew formulae, and may need to be deleted.
    
    Unexpected header files:
      /usr/local/include/python3.8/python-igraph/igraphmodule_api.h

The system seems to be working fine, but naturally, I want to get rid of this error message.  Can you help me?

**Comment by carlocab** [comment]
Try `brew uninstall --zap mysqlworkbench`. **Note** the caveat:

    --zap  Remove all files associated with a cask. May remove files which are shared between applications.

**Reply by saulspatz** [reply]
Thanks.  `rm -rf $(brew --prefix)/Caskroom/mysqlworkbench` worked.  I wanted to close thsi discussion, but I couldn't figure out how.

**Reply by carlocab** [reply]
You can mark this answer as having answered your question. Or post a new answer with your latest comment and mark that as the answer.

**Reply by saulspatz** [reply]
@carlocab Dione.  Thank you.

**Comment by saulspatz** [comment, marked as answer]
`rm -rf $(brew --prefix)/Caskroom/mysqlworkbench` worked.  This formula was supplied to me [here](https://github.com/Homebrew/homebrew-cask/issues/97136#issuecomment-753020849) by Vítor Galvão
